### ISSUE TITLE
[8.2.0] If the beforeRowMove callback does `return true` instead of `return` it causes a crash in undo

### ISSUE THREAD

**Issue by henricook** [handsontable/handsontable#7497]
### Description
[8.2.0] If the beforeRowMove callback does `return true` instead of `return` it causes a crash in undo

This took a lot of my time to figure out that returning true was breaking everything! I'm still not sure why it makes a difference.

### Steps to reproduce
1. Open fiddle
2. Open dev console
3. Move a row
4. In failing fiddle: `rows.slice() is not a function` (unhandled exception)
5. In succeeding fiddle: works fine

### Demo
Failing fiddle (uses `return true`): https://jsfiddle.net/qb80e6zo/
Succeeding fiddle (uses `return`): https://jsfiddle.net/qb80e6zo/1/

### Your environment
* Handsontable version: 8.2.0
* Browser Name and version: Firefox 84.0.2
* Operating System: Ubuntu

**Comment by henricook** [comment]
Once I started looking for this pattern I also found that if I return true from the `beforeRemoveRow` callback it breaks Undo as well! Once I started doing `return` from there I could undo remove rows whereas before that it was doing some weird 'clone the first line on the table' behaviour

If there's some kind of expected return from this `before` callbacks it would be good to know (and document) what it is.

**Comment by wojciechczerniak** [comment]
Most of the `before*` hooks may return `undefined` to behave like an event, `false` to block the operation, or any other value to alter the value it transfers, like a middleware. Usually, it is the same as the first callback argument. 

So when you return `true` it tries to use it as an expected type for this particular hook and continue to work with the new value you have provided.

**Comment by henricook** [comment]
Thanks for clarifying @wojciechczerniak - how would i return all the arguments? I tried things like:

`return [rows, finalIndex, otherStuffHere]`

and

`return {rows, finalIndex, otherStuffHere]`

but that didn't seem to work

**Comment by AMBudnik** [comment]
Hey @henricook maybe https://jsfiddle.net/handsoncode/3de89nLx/ 

```js
beforeRowMove: function(...abc) {
      return abc
}
```

**Comment by henricook** [comment]
Assuming i'm destructing them, and am using functional components:

```javascript
beforeRowMove = (rows, finalIndex, otherStuff) => { 
 return ???
}
```

how would that look?

**Comment by AMBudnik** [comment]
I think that we went too far in explanation with that example. the hook doesn't need to be returned anything to perform the movement of rows. You just have the ability to block it while using `return false`.

Maybe we should add some info to the docs or even treat `return true` as the confirmation. Let me check it with the team.

**Comment by AMBudnik** [comment]
@kirszenbaum I just checked the issue using Handsontable v12.2.0 https://jsfiddle.net/pndcjhzx/ and it no longer breaks for `return true` so I'm not sure if we should add any additional info.

**Comment by AMBudnik** [comment]
Issue solved as done in 12.2.0.
Again thank you for the feedback.
